core/controlplane/config/config.go

```diff
@@ -1482,3 +1482,9 @@ func withHostedZoneIDPrefix(id string) string {
 	}
 	return id
 }
+
+func (c Config) AWSAccountID() string {
+	re := regexp.MustCompile("(?:[0-9]{12})")
```

Comment:
I guess templating is done before kube-aws contacts AWS, so it is too early to receive real AWS account name?

Reason this might not work in 0.00001% of cases is if KMS key is managed by security teams across accounts, but I think having kube2iam out of the box is more important :)